assets/src/block-validation/amp-validation-status/revalidate-notification.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { usePrevious } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { BLOCK_VALIDATION_STORE_KEY } from '../store';
+import BellIcon from '../../../images/bell-icon.svg';
+import { SidebarNotification } from '../../block-editor/components/sidebar-notification';
+
+/**
+ * AMP re-validate status message.
+ */
+export default function AMPRevalidateNotification() {
+	const [ didFetchErrors, setDidFetchErrors ] = useState( false );
+	const [ loadingMessage, setLoadingMessage ] = useState( '' );
+	const { autosave, savePost } = useDispatch( 'core/editor' );
+
+	const {
+		hasErrorsFromRemovedBlocks,
+		hasActiveMetaboxes,
+		isDraft,
+		isFetchingErrors,
+		isEditedPostNew,
+		isPostDirty,
+	} = useSelect( ( select ) => {
+		let _hasErrorsFromRemovedBlocks = false;
+		for ( const validationError of select( BLOCK_VALIDATION_STORE_KEY ).getValidationErrors() ) {
+			const { clientId } = validationError;
+			const blockDetails = clientId ? select( 'core/block-editor' ).getBlock( clientId ) : null;
+			if ( clientId && ! blockDetails ) {
```

Comment:
As in aa668d0, we could get here just the block name instead of the entire block object, i.e.

```suggestion
			if ( clientId && ! select( 'core/block-editor' ).getBlockName( clientId ) ) {
```

Reply:
When I think about it, we could actually replace the entire loop with `find` method like

```diff
-		let _hasErrorsFromRemovedBlocks = false;
-		for ( const validationError of select( BLOCK_VALIDATION_STORE_KEY ).getValidationErrors() ) {
-			const { clientId } = validationError;
-			const blockDetails = clientId ? select( 'core/block-editor' ).getBlock( clientId ) : null;
-			if ( clientId && ! blockDetails ) {
-				_hasErrorsFromRemovedBlocks = true;
-				break;
-			}
-		}
-
-		return {
-			hasErrorsFromRemovedBlocks: _hasErrorsFromRemovedBlocks,
+		const errors = select( BLOCK_VALIDATION_STORE_KEY ).getValidationErrors();
+
+		return {
+			hasErrorsFromRemovedBlocks: Boolean( errors.find( ( { clientId } ) => clientId && ! select( 'core/block-editor' ).getBlockName( clientId ) ) ),
```

But I guess it might not be as readable as the loop.

Reply:
I like that latter suggestion. Go ahead with it!

Reply:
Sounds good, it's done (in an even shorter form) in 17ef326.